cli/pkg/modelregistry/client.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	m, err := c.CreateRegisteredModel(modelName, modelDescription, nil)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	v, err := c.CreateModelVersion(m.GetId(), versionName, metaData)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	a, err := c.CreateModelArtifact(
+		v.GetId(), artifactName, modelDescription, uri, modelFormatName, modelFormatVersion, artifactExternalId,
+	)
+
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return m, v, a, nil
+}
+
+func (c *Client) CreateRegisteredModel(name string, description string, metaData map[string]openapi.MetadataValue) (
+	*openapi.RegisteredModel,
+	error,
+) {
+	if name == "" || description == "" {
+		return nil, fmt.Errorf("name and description are required")
+	}
+
+	m := openapi.NewRegisteredModelCreateWithDefaults()
+	m.SetName(name)
+	m.SetDescription(description)
+
+	if metaData != nil {
+		m.SetCustomProperties(metaData)
+	}
+
+	model, resp, err := c.modelRegistryClient.ModelRegistryServiceAPI.CreateRegisteredModel(context.Background()).
+		RegisteredModelCreate(*m).Execute()
+	if err != nil {
+		if resp == nil {
+			return nil, fmt.Errorf("error creating registered model: %w", err)
+		} else {
+			// Currently model registry returns 500 when the model exists. This is a workaround to handle the error
+			// until the model registry is fixed. The workaround is to check the error message and return ErrModelExists
+			// if the error message contains the expected error message.
+			// TODO: Remove this workaround when model registry returns 403 when the model exists
+			if resp.StatusCode != 201 && isOpenAPIErrorOfKind(err, ErrAlreadyExists) {
+				return nil, fmt.Errorf("%w. model name: %s", ErrModelExists, name)
+			}
+			// This is a weird case where we got a response and an error that we're unable to handle.
+			return nil, fmt.Errorf(
+				"error while creating a registered model: server responded with %s %w", resp.Status, err,
+			)
+		}
+	}
+	if resp.StatusCode != 201 {
+		return nil, fmt.Errorf("failed to create a registered model: %s", resp.Status)
+	}
+	return model, nil
+}
+
+func (c *Client) CreateModelVersion(
+	modelId string,
+	versionName string,
+	metaData map[string]openapi.MetadataValue,
+) (*openapi.ModelVersion, error) {
+
+	if modelId == "" || versionName == "" {
+		return nil, fmt.Errorf("model ID and version are required")
+	}
+
+	modelVersion := openapi.NewModelVersionWithDefaults()
+	modelVersion.SetName(versionName)
+
+	if metaData != nil {
+		modelVersion.SetCustomProperties(metaData)
+	}
+
+	v, resp, err := c.modelRegistryClient.ModelRegistryServiceAPI.CreateRegisteredModelVersion(
+		context.Background(), modelId,
+	).ModelVersion(*modelVersion).Execute()
+
+	if err != nil {
+		if resp == nil {
+			return nil, fmt.Errorf("error creating model version: %w", err)
+		} else {
+			if resp.StatusCode != 201 {
+				// TODO: Remove this workaround when model registry returns 404 when the model is not found
```

Comment:
Do you mean linked in the source code?

Reply:
Yeah, possibly. I think linking issues in the source code is useful, so that when we stumble across this TODO in future, we'll have concrete places to look to see if it can be progressed.